[D1] Rename --count to --limit in wrangler insights (#6464)
cloudflare/workers-sdk: .changeset/angry-steaks-wait.md

---

> "wrangler": major

Reviewer: ```suggestion
"wrangler": minor
```

---

> feat: rename `--count` to `--limit` in wrangler insights

Reviewer: ```suggestion
feat: rename `--count` to `--limit` in `wrangler d1 insights`
```

> This is a breaking change and was made to align with traditional naming conventions and improve clarity.

Reviewer: ```suggestion
BREAKING CHANGE: This PR renames `wrangler d1 insight`'s `--count` flag to `--limit` to improve clarity and conform to naming conventions.
```
